x/programs/examples/counter_test.go

```diff
@@ -95,41 +96,41 @@ func TestCounterProgram(t *testing.T) {
 	require.NoError(err)
 
 	// define max units to transfer to second runtime
-	unitsTransfer := uint64(10000)
+	unitsTransfer := uint64(20000)
 
 	// transfer the units from the original runtime to the new runtime before
 	// any calls are made.
 	_, err = rt.Meter().TransferUnitsTo(rt2.Meter(), unitsTransfer)
 	require.NoError(err)
 
 	// simulate creating second program transaction
-	program2ID := ids.GenerateTestID()
-	err = storage.SetProgram(ctx, db, program2ID, wasmBytes)
+	programID2 := ids.GenerateTestID()
+	err = storage.SetProgram(ctx, db, programID2, wasmBytes)
 	require.NoError(err)
 
 	mem2, err := rt2.Memory()
 	require.NoError(err)
-	programID2Ptr, err := argumentToSmartPtr(program2ID, mem2)
+	programID2Ptr, err := argumentToSmartPtr(programID2, mem2)
 	require.NoError(err)
 
 	// write alice's key to stack and get pointer
 	alicePtr2, err := argumentToSmartPtr(aliceKey, mem2)
 	require.NoError(err)
 
 	// initialize counter for alice on runtime 2
-	result, err = rt2.Call(ctx, "initialize_address", programID2Ptr, alicePtr2)
+	result, err = rt2.Call(ctx, "initialize_address", program.Context{ProgramID: programID2}, alicePtr2)
```

Comment:
So if I understand correctly, the `program.Context` is passed here, pre-populated with the `ProgramID` only.
Isn't it a bit redundant though ? According to the "old" way, you would pass the `programID` as the function parameter, and as of this feature, the `program.Context` field would all be populated with this `programID` in the `Call` function.

Reply:
@iFrostizz, it's easier to keep the function signatures the same and extend the context rather than having to change every signature every time we want to add to the context. This PR simply gives us a place where we can expand things without breaking old programs.

Reply:
Thanks, that makes sense !